apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletApiAdvice.java

```diff
@@ -230,10 +238,23 @@ public static <HttpServletRequest, HttpServletResponse, ServletContext, ServletC
                     final int size = requestExceptionAttributes.size();
                     for (int i = 0; i < size; i++) {
                         String attributeName = requestExceptionAttributes.get(i);
-                        Object throwable = adapter.getHttpAttribute(httpServletRequest, attributeName);
+                        Object throwable = adapter.getAttribute(httpServletRequest, attributeName);
                         if (throwable instanceof Throwable) {
                             t2 = (Throwable) throwable;
-                            if (!attributeName.equals("javax.servlet.error.exception") && !attributeName.equals("jakarta.servlet.error.exception")) {
+
+                            // elastic exception can be removed as it's not needed after transaction end
+                            if (attributeName.equals(ELASTIC_EXCEPTION)) {
+                                try {
+                                    // extra safety: try to remove hard reference from the map entry to the exception
+                                    // but map implementation might throw if it does not support null values
+                                    adapter.setAttribute(httpServletRequest, attributeName, null);
```

Comment:
So if we know that `ConcurrentHashMap` will always throw here and we are trying to circumvent the problem exactly for `ConcurrentHashMap` I wonder if we want this cost of the extra safety:

How about we go with your initial suggest and just do the remove without trying to set it to `null`?

Reply:
Yes, I think that's a better option here, the removal is expected to remove the complete map entry.

I'll revert to only remove.